File: java/sfo_62_josephus_problem_s1/sfo_62_josephus_problem_s1.java
/*
* File: sfo_62_josephus_problem_s1.java
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

package sfo_62_josephus_problem_s1;

import include.*;
import java.util.*;

// ===== Solution Code =====
class Solution {
    public int lastRemaining(int n, int m) {
        int x = 0;
        for (int i = 2; i <= n; i++) {
            x = (x + m) % i;
        }
        return x;
    }
}

public class sfo_62_josephus_problem_s1 {
    public static void main(String[] args) {
        // ======= Test Case =======
        int n = 5;
        int m = 3;
        // ====== Driver Code ======
        Solution slt = new Solution();
        int res = slt.lastRemaining(n, m);
        System.out.println(res);
    }
}

File: python/sfo_62_josephus_problem_s1.py
'''
File: sfo_62_josephus_problem_s1.py
Created Time: 2021-12-09
Author: Krahets (krahets@163.com)
'''

from include import *

# ===== Solution Code =====
class Solution:
    def lastRemaining(self, n: int, m: int) -> int:
        x = 0
        for i in range(2, n + 1):
            x = (x + m) % i
        return x

# ======= Test Case =======
n = 5
m = 3
# ====== Driver Code ======
slt = Solution()
res = slt.lastRemaining(n, m)
print(res)
